airflow/models/baseoperator.py

```diff
@@ -634,6 +650,43 @@ def deps(self) -> Set[BaseTIDep]:
             NotPreviouslySkippedDep(),
         }
 
+    def _set_xcomargs_dependencies(self) -> None:
+        """
+        Resolves upstream dependencies of a task. In this way passing an ``XComArg``
+        as value for a template field will result in creating upstream relation between
+        two tasks.
+
+        **Example**: ::
+
+            with DAG(...):
+                generate_content = GenerateContentOperator(task_id="generate_content")
+                send_email = EmailOperator(..., html_content=generate_content.output)
+
+            # This is equivalent to
+            with DAG(...):
+                generate_content = GenerateContentOperator(task_id="generate_content")
+                send_email = EmailOperator(
+                    ..., html_content="{{ task_instance.xcom_pull('generate_content') }}"
+                )
+                generate_content >> send_email
+
+        """
+        from airflow.models.xcom_arg import XComArg
+
+        def apply_set_upstream(arg: Any):
+            if isinstance(arg, XComArg):
+                self.set_upstream(arg.operator)
+            elif isinstance(arg, (tuple, set, list)):
+                for elem in arg:
+                    apply_set_upstream(elem)
+            elif isinstance(arg, dict):
```

Comment:
There's also the custom "decorated" classes which are walked by template rendering

Reply:
@ash do you mean named tuples?

Reply:
https://github.com/apache/airflow/commit/d567f9ab8d5fdd6d18509fd8d623b26795eca25c

Reply:
@turbaszek 

You  haven't dealt with this case yet have you?

```python

class MyDataReader:
    template_fields = ['path']
    def __init__(self, my_path):
      self.path = my_path


with dag:
  generate_content = GenerateContentOperator(task_id="generate_content")
  t = PythonOperator(
      task_id='transform_data',
      python_callable=transform_data
      op_args=[
        MyDataReader(generate_content.output)
      ],
)
```

Reply:
@ashb sorry, I thought I did this last week. Fixed